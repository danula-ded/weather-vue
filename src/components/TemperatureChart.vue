<script setup lang="ts">
import {
  Chart as ChartJS,
  LineController,
  LineElement,
  PointElement,
  LinearScale,
  Title,
  CategoryScale,
  Tooltip,
  Legend,
} from 'chart.js';
import { Line } from 'vue-chartjs';
import { computed } from 'vue';
import type { TemperatureChartProps } from '@/types/chart';

ChartJS.register(
  LineController,
  LineElement,
  PointElement,
  LinearScale,
  Title,
  CategoryScale,
  Tooltip,
  Legend
);

const props = defineProps<TemperatureChartProps>();

const chartData = computed(() => ({
  labels: props.labels,
  datasets: [
    {
      label: 'Temperature (°C)',
      data: props.temperatures,
      borderColor: 'var(--primary-color)',
      backgroundColor: 'rgba(102, 126, 234, 0.1)',
      borderWidth: 3,
      fill: true,
      tension: 0.4,
      pointBackgroundColor: 'var(--primary-color)',
      pointBorderColor: '#ffffff',
      pointBorderWidth: 2,
      pointRadius: 6,
      pointHoverRadius: 8,
      pointHoverBackgroundColor: 'var(--secondary-color)',
      pointHoverBorderColor: '#ffffff',
    },
  ],
}));

const chartOptions = {
  responsive: true,
  maintainAspectRatio: false,
  plugins: {
    legend: {
      display: false,
    },
    tooltip: {
      backgroundColor: 'var(--bg-glass)',
      titleColor: 'var(--text-primary)',
      bodyColor: 'var(--text-secondary)',
      borderColor: 'rgba(255, 255, 255, 0.1)',
      borderWidth: 1,
      cornerRadius: 8,
      displayColors: false,
      titleFont: {
        size: 14,
        weight: '600',
      },
      bodyFont: {
        size: 13,
      },
      padding: 12,
      callbacks: {
        title: (context: any) => {
          return `Time: ${context[0].label}`;
        },
        label: (context: any) => {
          return `Temperature: ${context.parsed.y}°C`;
        },
      },
    },
  },
  scales: {
    x: {
      grid: {
        color: 'rgba(255, 255, 255, 0.05)',
        drawBorder: false,
      },
      ticks: {
        color: 'var(--text-secondary)',
        font: {
          size: 12,
        },
        maxRotation: 45,
      },
      border: {
        color: 'rgba(255, 255, 255, 0.1)',
      },
    },
    y: {
      grid: {
        color: 'rgba(255, 255, 255, 0.05)',
        drawBorder: false,
      },
      ticks: {
        color: 'var(--text-secondary)',
        font: {
          size: 12,
        },
        callback: (value: any) => `${value}°C`,
      },
      border: {
        color: 'rgba(255, 255, 255, 0.1)',
      },
    },
  },
  interaction: {
    intersect: false,
    mode: 'index' as const,
  },
  elements: {
    point: {
      hoverRadius: 8,
    },
  },
};
</script>

<template>
  <div class="chart-container">
    <Line :data="chartData" :options="chartOptions" />
  </div>
</template>

<style scoped>
.chart-container {
  position: relative;
  height: 300px;
  width: 100%;
}

@media (max-width: 768px) {
  .chart-container {
    height: 250px;
  }
}

@media (max-width: 480px) {
  .chart-container {
    height: 200px;
  }
}
</style>
