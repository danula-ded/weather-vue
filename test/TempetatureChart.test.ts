import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/vue';
import TemperatureChart from '@/components/TemperatureChart.vue';

describe('TemperatureChart.vue', () => {
  it('renders canvas element', () => {
    const { getByRole } = render(TemperatureChart, {
      props: {
        labels: ['10:00', '11:00', '12:00'],
        temperatures: [21, 22, 23],
      },
    });

    const canvas = getByRole('img');
    expect(canvas).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { container } = render(TemperatureChart, {
      props: {
        labels: ['13:00', '14:00', '15:00'],
        temperatures: [25, 26, 27],
      },
    });

    expect(container).toMatchSnapshot();
  });
});
