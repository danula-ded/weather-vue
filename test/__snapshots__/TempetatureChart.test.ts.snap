// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TemperatureChart.vue > matches snapshot 1`] = `
<div>
  <div
    class="chart-container"
    data-v-3234bcc4=""
  >
    <canvas
      data-v-3234bcc4=""
      role="img"
    >
      <p>
        
      </p>
    </canvas>
  </div>
</div>
`;
